README.md

```diff
@@ -19,113 +19,122 @@ This library includes the following modules.
 - `magnolify-guava` - type class derivation for [Guava](https://guava.dev)
-  - [`Funnel[T]`](https://guava.dev/releases/25.0-jre/api/docs/com/google/common/hash/Funnel.html)
-  
-- `magnolify-avro` - conversion between Scala case classes and [Apache Avro](https://github.com/apache/avro) `GenericRecord`
-- `magnolify-bigquery` - conversion between Scala case classes and [Google Cloud BigQuery](https://cloud.google.com/bigquery/) `TableRow`
-- `magnolify-datastore` - conversion between Scala case classes and [Google Cloud Datastore](https://cloud.google.com/datastore/) `Entity`
-- `magnolify-tensorflow` - conversion between Scala case classes and [TensorFlow](https://www.tensorflow.org/) `Example`
+  - [`Funnel[T]`](https://guava.dev/releases/snapshot-jre/api/docs/com/google/common/hash/Funnel.html)
+
+- `magnolify-avro` - conversion between Scala types and [Apache Avro](https://github.com/apache/avro) `GenericRecord`
+- `magnolify-bigquery` - conversion between Scala types and [Google Cloud BigQuery](https://cloud.google.com/bigquery/) `TableRow`
+- `magnolify-datastore` - conversion between Scala types and [Google Cloud Datastore](https://cloud.google.com/datastore/) `Entity`
+- `magnolify-tensorflow` - conversion between Scala types and [TensorFlow](https://www.tensorflow.org/) `Example`
 
 # Usage
 
 Cats, ScalaCheck, and Guava type class derivation can be performed both automatically and semi-automatically.
 
-Automatic derivation with implicits:
+Automatic derivation works out of the box by importing implicits from `magnolify.$module.auto._`.
+It works with context bounds i.e. `def plus[T: Semigroup](x: T, y: T)` and implicit paramemters i.e.
+`def f[T](x: T, y: T)(implicit sg: Semigroup[T])`.
+
 ```scala
-case class MyCaseClass(field: NestedCaseClass) 
-case class NestedCaseClass(innerField: String) // works with other non-String elements, this is just an example
+case class Inner(int: Int, str: String)
+case class Outer(inner: Inner)
 
-// implicitly keyword summons an instance of the typeclass wrapping a given case class
-// typeclasses are generated at compile time with macros & Magnolia
-// this works if each element within the case class has a typeclass instance defined within the implicit scope
+// implicitly[T] summons an implicit instance of T
+// - which Magnolia derives at compile time if T is a typeclass F[A]
+// - and A is a case class or ADT
+// - recursively summons typeclasses for case class fields or ADT subtypes
 
-// an example with Cats using the Eq typeclass
+// an example with Cats Semigroup
 import magnolify.cats.auto._
-import cats.kernel.Eq
-// similar to Eq.fromUniversalEquals for a case class but it's at compile time, and don't need == defined reasonably
-// instead of using == in fromUniversalEquals, you need an Eq instance for each part of your case class
-implicit val eqString: Eq[String] = ??? // could be fromUniversalEquals, but doesn't have to be
```

Comment:
It's hard to explain `Eq` here, and it's outside the scope and those who need it should read cats doc.
`Semigroup` is more example friendly.